eagle-core/eagle-policy/eagle-policy-base/src/main/java/org/apache/eagle/policy/dao/PolicyDefinitionEntityDAOImpl.java

```diff
@@ -17,14 +17,17 @@
 package org.apache.eagle.policy.dao;
 
 import org.apache.eagle.alert.entity.AbstractPolicyDefinitionEntity;
+import org.apache.eagle.alert.entity.AlertDefinitionAPIEntity;
```

Comment:
The updatePolicyDetails() already uses T for getting the entity. 
The logic inside of casting to AlertDefinitionAPIEntity and adding values specific to the class has been moved outside the impl class.